### PR TITLE
MBS-9389 Prepare to support API 30

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
Basic Android 11 support

- New emulator image
- Use publicly accessible dir for artifacts to overcome scoped storage restrictions
In Android 11 we can't pull from application internal dir. 
Didn't [disable scoped storage restrictions](https://developer.android.com/training/data-storage/use-cases#opt-out-in-tests) to be able to test it properly.

### CODE REVIEW

File: samples/test-runner/build.gradle.kts [hunk 1]
```diff
@@ -58,6 +58,15 @@ instrumentation {
             cpuCoresLimit = "1.3",
             memoryLimit = "4Gi"
         )
```
Comment (eugene-krivobokov) [inline samples/test-runner/build.gradle.kts:60]: Didn't rebuild older images.
I've checked locally that changes in Docker didn't affect them: qemu version is the same, works locally.

File: samples/test-runner/build.gradle.kts [hunk 2]
```diff
@@ -84,6 +93,23 @@ instrumentation {
...
+                            }
+
+                            testsCountBasedReservation {
+                                device = emulator30
+                                maximum = 1
+                                minimum = 1
+                                testsPerEmulator = 1
+                            }
+                        }
+                    }
```
Comment (eugene-krivobokov) [inline samples/test-runner/build.gradle.kts:112]: Added to sample app to check basic integration.
Can remove after we finish update to 30 API.

File: subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt [hunk 3]
```diff
@@ -34,20 +34,24 @@ public object TestArtifactsProviderFactory {
...
+        api: Int,
         appUnderTestPackage: String,
         name: TestName,
     ): TestArtifactsProvider {
-        val dataPath = "/sdcard/Android/data/$appUnderTestPackage/files"
+        val dataPath = if (api >= 30) {
+            "/storage/emulated/0/Android/media/$appUnderTestPackage"
+        } else {
+            "/sdcard/Android/data/$appUnderTestPackage/files"
+        }
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt:48]: Need advice on hiding these implementation details.
This knowledge is on different layers.
Glad to discuss it on call.

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/listener/ArtifactsTestListener.kt [hunk 4]
```diff
@@ -116,6 +104,51 @@ internal class ArtifactsTestListener(
...
+            hostDir = destination,
+            validator = reportArtifactsPullValidator
+        )
+        return if (device.api >= 30) {
+            val cleaningResult = device.clearDirectory(remotePath)
+                .map { remotePath.toFile() }
+
+            pullResult.combine(cleaningResult) { pullDir, _ ->
+                pullDir
+            }
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/listener/ArtifactsTestListener.kt:146]: Don't want to switch to new behavior for all APIs.
It costs an additional useless adb invocation - more time and reasons to fail.
`pm clear` do the job in previous behavior.

File: subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt:48]: yes, let's discuss it after tomorrow's sync call

**Review by dsvoronin — APPROVED** [review]

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/listener/ArtifactsTestListener.kt [hunk 5]
```diff
@@ -116,6 +104,51 @@ internal class ArtifactsTestListener(
...
+    }
+
+    private fun pullResults(device: Device, remotePath: Path, destination: Path): Result<File> {
+        val pullResult = device.pullDir(
+            deviceDir = remotePath,
+            hostDir = destination,
+            validator = reportArtifactsPullValidator
+        )
+        return if (device.api >= 30) {
+            val cleaningResult = device.clearDirectory(remotePath)
```
Comment (eugene-krivobokov) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/listener/ArtifactsTestListener.kt:141]: this is a responsibility of DeviceWorker

UPD: fixed here https://github.com/avito-tech/avito-android/pull/1156

File: subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt (hunk 3 above)
Comment (eugene-krivobokov) [reply subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt:48]: https://github.com/avito-tech/avito-android/pull/1156

**Pull request merged**